### PR TITLE
add amazon polarity dataset

### PR DESCRIPTION
**Pull request by hfawaz**
This corresponds to the amazon (binary dataset) requested in https://github.com/huggingface/datasets/issues/353

### CODE REVIEW

**Review by lhoestq — COMMENTED** [review]
Awesome thanks !

Could you also add the tags in the dataset card please ?
More info here : https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#tag-the-dataset-and-write-the-dataset-card

Also what's the difference with this dataset : https://github.com/huggingface/datasets/tree/master/datasets/amazon_us_reviews ?

**Comment by hfawaz** [comment]
`amazon_polarity` is probably a subset `amazon_us_reviews` but I am not entirely sure about that.
I guess `amazon_polarity` will help in reproducing results since even if it is a subset from `amazon_us_reviews`, it is not trivial how to extract `amazon_polarity` from `amazon_us_reviews`, especially since `amazon_us_reviews` was released after `amazon_polarity`.

**Comment by hfawaz** [comment]
do you know what the problem would be ? should I pull the master before ? @lhoestq

**Comment by lhoestq** [comment]
The error just appeared on master. I will try to fix it today.
You can ignore them since it's not related to the dataset you added

File: datasets/amazon_polarity/README.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+multilinguality:
+- monolingual
+size_categories:
+- n>1M
+source_datasets:
+- original
+task_categories:
+- text-classification
+task_ids:
+- sentiment-classification
```
Comment (lhoestq) [inline datasets/amazon_polarity/README.md:18]:
```suggestion
---
annotations_creators:
- crowdsourced
language_creators:
- crowdsourced
languages:
- en
licenses:
- apache-2.0
multilinguality:
- monolingual
size_categories:
- n>1M
source_datasets:
- original
task_categories:
- text-classification
task_ids:
- sentiment-classification
---
```

**Review by lhoestq — APPROVED** [review]
Thanks for adding the tags :) 

LGTM now

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Comment by hfawaz** [comment]
Great thanks for the help.

**Pull request merged**